src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java

```diff
@@ -113,14 +114,16 @@ public SampleEndpoint(SampleService sampleService, MapperFacade mapperFacade) {
    * @return a newly created sample summary that won't have samples or collection instruments populated
    * @throws CTPException thrown if upload started message cannot be sent
    */
-  public SampleSummary ingest(final MultipartFile file, final String type) throws CTPException {
+  public SampleSummary ingest(final MultipartFile file, final String type, String filename) throws CTPException {
     final SampleSummary newSummary = this.sampleService.createAndSaveSampleSummary();
 
     Callable<Optional<SampleSummary>> callable =() -> {
       try {
         return Optional.of(this.sampleService.ingest(newSummary, file, type));
+      } catch (IngesterCTPException e){
+        return this.sampleService.failSampleSummary(newSummary, e.getErrorCode(), e);
       } catch (Exception e) {
-        return this.sampleService.failSampleSummary(newSummary, e);
+        return this.sampleService.failSampleSummary(newSummary, SampleSummaryDTO.ErrorCode.NotSpecified, e);
```

Comment:
We'll probably want to log this exception I guess?

Reply:
This gets logged in failSampleSummary